File: backend/ScrumGroup/ScrumGroup/Data/StudyGuide.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ScrumGroup.Data;

public class StudyGuide
{
    [Key]
    public int GuideId { get; set; }
    [ForeignKey("ReligionId")]
    public int ReligionId { get; set; }
    public Religion Religion { get; set; }
    [ForeignKey("UserId")]
    public int UserId { get; set; }
    public User? User { get; set; }
    public string? Response1 { get; set; }
    public string? Response2 { get; set; }
    public string? Response3 { get; set; }
    public string? Response4 { get; set; }
    public string? Response5 { get; set; }
}

File: frontend/src/types/StudyGuide.ts
import { Religion } from './Religion';

import { User } from './User';

export interface StudyGuide {
  guideId: number;
  religionId: number;
  religion: Religion; // Include the full Religion object
  UserId: number;
  User: User; // Include the full User object
  VideoSrc: string | null; // Handle nullable values
  Response1: string | null;
  Response2: string | null;
  Response3: string | null;
  Response4: string | null;
  Response5: string | null;
}
